aptos-move/framework/aptos-stdlib/sources/table_with_length.move

```diff
@@ -0,0 +1,82 @@
+/// Extends Table and provides functions such as length and the ability to be desttroyed
+
+module aptos_std::table_with_length {
+    use std::error;
+    use aptos_std::table::{Self, Table};
+
+    // native code raises this with error::invalid_arguments()
+    const EALREADY_EXISTS: u64 = 100;
+    // native code raises this with error::invalid_arguments()
+    const ENOT_FOUND: u64 = 101;
+    const ENOT_EMPTY: u64 = 102;
+
+    /// Type of tables
+    struct TableWithLength<phantom K: copy + drop, phantom V> has store {
```

Comment:
the name is not very .. convenient

Reply:
Do we really need this or can we just drop it? Users can do it themselves if the need to.

Reply:
We use it in token_transfer, which itself is not parallel. We can consider alternatives for that framework later, but it is important in that we can't (and don't want to) support unilateral transfers of tokens.

Reply:
You can't just wrap a table together with your own length field?

Reply:
We could, but then we have to deal with the ambiguity around destroying a table and unknown aspects in storage.